File: 62.unique-paths.py
# @leet start
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        ans = 1
        for k in range(1, min(m, n)):
            ans = ans * (m + n - 1 - k) // k
        return ans

# @leet end

File: 62.unique-paths.cpp
// @leet start
class Solution {
public:
  int uniquePaths(int m, int n) {
    long int ans = 1;
    int max = std::min(n, m);
    for (int i = 1; i < max; i++) {
      ans = ans * (m + n - 1 - i) / i;
    }
    return ans;
  }
};
// @leet end
